### ISSUE TITLE
File containing feature module is missing from the Typescript compilation - random error

### ISSUE THREAD

**Issue by gcharis** [angular/angular#28198]
# 🐞 bug report

### Description

Typescript compiler <i><b>sometimes</b></i> does not seem to compile files containing lazy loaded modules. It is a random encounter because minor changes to force the compiler to run again do seem to fix the issue.

Also there is no 'files' or 'include' property in the tsconfig file because I want to include all files in the containing directory and sub directories.

## 🔥 Exception or Error
<pre><code>
ERROR in path/to/lazyloaded.module.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/index.js):
Error: path/to/lazyloaded.module.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
    "@angular/animations": "~7.1.0",
    "@angular/common": "~7.1.0",
    "@angular/compiler": "~7.1.0",
    "@angular/core": "~7.1.0",
    "@angular/forms": "~7.1.0",
    "@angular/platform-browser": "~7.1.0",
    "@angular/platform-browser-dynamic": "~7.1.0",
    "@angular/router": "~7.1.0",
    "@angular-devkit/build-angular": "^0.11.4",
    "@angular-devkit/build-ng-packagr": "~0.11.0",
    "@angular/cli": "~7.1.2",
    "@angular/compiler-cli": "~7.1.0",

</code></pre>

## Edit

I'm providing a <a href="https://github.com/gcharis/compilation-error-example/tree/master/projects"> github repo</a> that simulates my project. 

I have an angular library named foo-lib that handles its routing and lazy load one of its modules, i.e super module. The application that wraps this library does not interfere with the routing system.

Running ng serve produces the above error while running ng serve --aot produces the following error:

<pre><code>
ERROR in ./$$_lazy_route_resource lazy namespace object
Module not found: Error: Can't resolve 'path/to/projects/foo-lib/src/lib/super/super.module.ngfactory.js' in 'path\to\projects\foo-tester\$$_lazy_route_resource'
</code></pre>

**Comment by mohammedzamakhan** [comment]
Should this be in https://github.com/angular/angular-cli repo?

**Comment by gcharis** [comment]
I don't think I understand exactly your post. I updated my initial post with a <a href="https://github.com/gcharis/compilation-error-example">github repo</a>. Maybe this has to do with the angular library handling the app routing?

**Comment by mhevery** [comment]
Closing as:
> Looks like dup of [angular/angular-cli#11203](https://github.com/angular/angular-cli/issues/11203)

**Comment by gcharis** [comment]
Ok after a lot of research and tweaks on the code I found a workaround. Just import your lazy loaded module inside your app. My guess is the following. 
<b><i>The lib is responsible for the routing system while the app loads the lib router. That creates conflicts for the file path that contains the lazy loaded module and eventually leads to the fact that the compiler does not have a reference to the desired module. I guess just by importing that module - without using it directly at all - does create a reference so that the compiler can compile it.</i></b>

The above repo has been updated with said workaround.
